crates/compilers/src/compilers/zksolc/settings.rs

```diff
@@ -66,6 +85,12 @@ pub struct ZkSettings {
     /// Whether to compile via EVM assembly.
     #[serde(default, rename = "forceEVMLA")]
     pub force_evmla: bool,
+    /// Suppressed `zksolc` warnings.
+    #[serde(default, skip_serializing_if = "HashSet::is_empty")]
+    pub suppressed_warnings: HashSet<ZkSolcWarning>,
```

Comment:
Alternatively these fields could go on `ZkSolcSettings` vs here in order to remove duplicating the fields in both `ZkSettings` and `ZkSolcInput`. Then when building the input we pass them directly

Reply:
IIUC, I should not have put any new settings outside of the `settings` field.
Would it help if I deprecated them where they currently are and moved them inside `settings`?

Reply:
@hedgar2017 that does seem cleaner

Reply:
@elfedy please review: https://github.com/matter-labs/era-compiler-solidity/pull/176
Do you need a stable release for testing, or a pre-release would be enough?

Reply:
nice, we can use a pre-release and specify a path to test

Reply:
@elfedy please try [this one](https://github.com/matter-labs/era-compiler-solidity/actions/runs/11451021369)!

Reply:
@slowli @hedgar2017 tested the new release here: https://github.com/Moonsong-Labs/compilers/commit/b04a92034e978a1626fd81b2a9fdc335f42e38e8 (Replace `<PATH_TO_RELEASE>` with the local path to zksolc test release). Seems to be working correctly (Tests pass with it and fail with previous releases)

Reply:
Maybe we can wait until those changes are part of a stable release then bump the version here and incorporate the changes in that test commit.

Reply:
@elfedy great!
We're going to make a release in the next few days.

Reply:
@elfedy released with zksolc v1.5.7!
Also supported by zksolc v1.5.6 and older, but one level above the `settings` object.